simplerunner.lua

```diff
@@ -315,23 +335,69 @@ function runner.run_benchmark_list(benchmarks, count, options)
 
     local package_path = package.path
     local package_cpath = package.cpath
-    
+    local results = {}
+
     for i, name in ipairs(benchmarks) do
-        local times = runner.runbench(name, count, options.scaling or scaling[name])
-        local stats = runner.calculate_stats(times)
-        print(string.format("  Mean: %f +/- %f, min %f, max %f", stats.mean, stats.cinterval, stats.min, stats.max))
-        if jitstats then
-            jitstats.print()
+        local scaling = options.scaling or scaling[name] or 1
+        local stats
+
+        if options.inprocess then
+            local times, jstats = runner.runbench(name, count, scaling)
+            stats = runner.calculate_stats(times)
+            stats.times = times
+            print("  " .. runner.fmtstats(stats))
+            if jitstats then
+                stats.jitstats = jstats
+                jitstats.print()
+            end
+        else
+           runner.runbench_outprocess(name, count, scaling, options)
         end
-        -- Try to clean away the current benchmark and its data, so the behaviour of the GC is more predictable for the next benchmark.
-        collectgarbage("collect")
 
-        -- Restore the Lua module search path since we set it to the directory of the benchmark when we load it.
-        package.path = package_path
-        package.cpath = package_cpath
+        if options.inprocess then
+            -- Try to clean away the current benchmark and its data, so the behaviour of the GC is more predictable for the next benchmark.
+            collectgarbage("collect")
+
+            -- Restore the Lua module search path since we set it to the directory of the benchmark when we load it.
+            package.path = package_path
+            package.cpath = package_cpath
+        end
     end
 end
 
+function runner.runbench_outprocess(name, count, scaling, options)
+    local lc = require("luachild")
+    local read, write = lc.pipe()
+
+    local p = lc.spawn{
+        command = arg[-1],
+        args = {
+            arg[0], "@RUNBENCH@", name, count, scaling, unpack(arg, 1) 
```

Comment:
Is `@RUNBENCH@` a magic marker? Wouldn't it be cleaner to have a "hidden" option "--child" or something like that?

Reply:
It made things simpler to have those arguments in fixed position they don't make any sense in the parent process but i could just change @RUNBENCH@ to --child or --runbench but it would still just be a marker. I also branch on it to initialize the process differently in the child process

Reply:
I'm fine with it being in a fixed position, but I think having it as a `--` option (whatever we call it) will be cleaner. [I'm also nervous about how some weird shell will interpret `@` in a file name!]

Reply:
i changed it to --childprocess